runtime/lua/vim/lsp.lua

```diff
@@ -938,7 +938,7 @@ end
 ---@param params table? Parameters to send to the server
 ---@param timeout_ms integer? Maximum time in milliseconds to wait for a result.
 ---                           (default: `1000`)
----@return table<integer, {err: lsp.ResponseError, result: any}>? result Map of client_id:request_result.
+---@return table<integer, {error: lsp.ResponseError, result: any}>? result Map of client_id:request_result.
```

Comment:
https://github.com/neovim/neovim/pull/28720#pullrequestreview-2051491767:

```suggestion
---@return table<integer, {error?: lsp.ResponseError, result: any}>? result Map of client_id:request_result.
```

Reply:
This change seems also needed for `vim.lsp.buf_request_all()` and the two `results` variables in `vim.lsp.buf.typehierarchy()`.